plugin/src/main/java/com/denizenscript/denizen/events/block/TNTPrimesScriptEvent.java

```diff
@@ -0,0 +1,77 @@
+package com.denizenscript.denizen.events.block;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.TNTPrimeEvent;
+
+public class TNTPrimesScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // tnt primes
+    //
+    // @Location true
+    //
+    // @Group Block
+    //
+    // @Cancellable true
+    //
+    // @Triggers when TNT is activated and will soon explode.
+    //
+    // @Context
+    // <context.entity> returns the entity that primed the TNT, if any.
+    // <context.block> returns the location of the block that primed the TNT, if any.
+    // <context.location> returns the location of the TNT block being primed.
+    // <context.cause> returns the cause of the TNT being primed. Refer to <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/block/TNTPrimeEvent.PrimeCause.html>
+    // <context.reason> deprecated in favor of <context.cause> for versions below 1.19.
+    //
+    // @Player when the priming entity is a player.
+    //
+    // -->
+
+    public TNTPrimesScriptEvent() {
+        registerCouldMatcher("tnt primes");
+    }
+
+    public TNTPrimeEvent event;
+    public LocationTag location;
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        if (!runInCheck(path, location)) {
+            return false;
+        }
+        return super.matches(path);
+    }
+
+    @Override
+    public ScriptEntryData getScriptEntryData() {
+        return new BukkitScriptEntryData(event.getPrimingEntity());
+    }
+
+    @Override
+    public ObjectTag getContext(String name) {
+        return switch (name) {
+            case "entity" -> event.getPrimingEntity() != null ? new EntityTag(event.getPrimingEntity()).getDenizenObject() : super.getContext(name);
+            case "block" -> event.getPrimingBlock() != null ? new LocationTag(event.getPrimingBlock().getLocation()) : super.getContext(name);
+            case "location" -> location;
+            case "cause" -> new ElementTag(event.getCause());
+            case "reason" -> new ElementTag(event.getCause().name().equals("PLAYER") ? "ITEM" : event.getCause().name()); // For backwards-compatibility

```

Comment:
Compare the enum itself, something like `== TNTPrimeEvent.PrimeCause.PLAYER`

Reply:
Also, if you use the String constructor should make it plain-text (which the enum constructor does automatically) - I.e.  just pass in `true` to the plain-text constructor